js/private/js_run_devserver.mjs

```diff
@@ -48,10 +48,26 @@ async function syncRecursive(src, dst, writePerm) {
         const exists = synced.has(src) || fs.existsSync(dst)
         synced.set(src, lstat.mtimeMs)
         if (lstat.isSymbolicLink()) {
+            const srcWorkspacePath = src.slice(RUNFILES_ROOT.length + 1)
             if (process.env.JS_BINARY__LOG_DEBUG) {
-                console.error(
-                    `Syncing symlink ${src.slice(RUNFILES_ROOT.length + 1)}`
+                console.error(`Syncing symlink ${srcWorkspacePath}`)
+            }
+            if (path.basename(path.dirname(src)) == 'node_modules') {
+                // Special case for node_modules symlinks where we should _not_ symlink to the runfiles but rather
+                // the bin copy of the symlink to avoid finding npm packages in multiple node_modules trees
+                const maybeBinSrc = path.join(
+                    process.env.JS_BINARY__EXECROOT,
+                    process.env.JS_BINARY__BINDIR,
+                    srcWorkspacePath
                 )
+                if (fs.existsSync(maybeBinSrc)) {
```

Comment:
Most of the fs ops in this file are the `async` versions  which I think would be np here as well?

Reply:
Would be a very minor optimization. Could do in a follow-up if we feel its worth it.

Reply:
Switching to use fs.access() to check existence here since that is what is recommended in the Node docs as exists is deprecated. Tho I'll keep existsSync for now for mkdirSync that is intentionally non-yielding and atomic.